File: backend/src/auth/dto/request/RegisterRequest.ts
import { Z } from 'zod-class';
import { AuthValidation } from '../auth.validation';
import { ApiProperty } from '@nestjs/swagger';

export class RegisterRequest extends Z.class(AuthValidation.REGISTER.shape) {
  @ApiProperty({
    description: 'User email',
    example: 'test@example.com',
  })
  email: string;

  @ApiProperty({
    description: 'User password',
    example: 'Password123!',
  })
  password: string;

  @ApiProperty({
    description: 'User name',
    example: 'John Doe',
  })
  name: string;

  @ApiProperty({
    description: 'User username',
    example: 'john_doe',
  })
  username: string;
}

File: mobile/packages/my_social_sdk/lib/models/register_request.dart
class RegisterRequest {
  final String email;
  final String password;
  final String name;
  final String username;

  RegisterRequest({
    required this.email,
    required this.password,
    required this.name,
    required this.username,
  });

  dynamic toJson() {
    return {
      'email': email,
      'password': password,
      'name': name,
      'username': username,
    };
  }
}
